packages/@aws-cdk/aws-sns/lib/subscription.ts

```diff
@@ -86,16 +102,48 @@ export class Subscription extends Resource {
       }
     }
 
+    this.deadLetterQueue = this.buildDeadLetterQueue(props);
+
     new CfnSubscription(this, 'Resource', {
       endpoint: props.endpoint,
       protocol: props.protocol,
       topicArn: props.topic.topicArn,
       rawMessageDelivery: props.rawMessageDelivery,
       filterPolicy: this.filterPolicy,
       region: props.region,
+      redrivePolicy: this.buildDeadLetterConfig(this.deadLetterQueue),
     });
 
   }
+
+  private buildDeadLetterQueue(props: SubscriptionProps) {
+    if (!props.deadLetterQueue) {
+      return undefined;
+    }
+
+    const deadLetterQueue = props.deadLetterQueue;
+
+    deadLetterQueue.addToResourcePolicy(new PolicyStatement({
```

Comment:
We should try to use [`queue.grantSendMessages`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-sqs.IQueue.html#grant-wbr-send-wbr-messagesgrantee). This can do more than just add the `sqs:SendMessage` permissions. If the queue is encrypted, it will also need key access. We should be able to use arnPrincipal to do this
```typescript
deadLetterQueue.grantSendMessages(new iam.ArnPrincipal(props.topic.topicArn));
```

Reply:
i wanted to do this, but this explicitly sets the ```Principal``` in the ```QueuePolicy``` to the SNS Topic. 

i'm not sure if that works, it looks like the docs expect ```Principal``` to be ```*``` https://aws.amazon.com/premiumsupport/knowledge-center/sqs-sns-subscribe-cloudformation/ or ```sns.amazonaws.com```, as done in https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-sns-subscriptions/lib/sqs.ts#L40-L47

i could be wrong, so i'll update the PR with your suggestion and then do some looking into it async